src/main/java/com/commercetools/pspadapter/tenant/TenantConfigFactory.java

```diff
@@ -0,0 +1,42 @@
+package com.commercetools.pspadapter.tenant;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@Component
+public class TenantConfigFactory {
+
+    private final Environment env;
+
+    @Autowired
+    public TenantConfigFactory(@Nonnull Environment env) {
+        this.env = env;
+    }
+
+    public Optional<TenantConfig> getTenantConfig(@Nonnull String tenantName) {
+        String projectKey = env.getProperty(tenantName + ".ctp.client.projectKey");
+        String clientId = env.getProperty(tenantName + ".ctp.client.clientId");
+        String clientSecret = env.getProperty(tenantName + ".ctp.client.clientSecret");
+        String ppPClientId = env.getProperty(tenantName + ".paypalPlus.client.clientId");
+        String ppPClientSecret = env.getProperty(tenantName + ".paypalPlus.client.clientSecret");
+        String ppPClientMode = env.getProperty(tenantName + ".paypalPlus.client.mode");
+
+        if (isAnyPropertyNull(projectKey, clientId, clientSecret,
+                ppPClientId, ppPClientSecret, ppPClientMode)) {
+            return Optional.empty();
```

Comment:
I don't want to put logging there because it would make multiple logging if tenant name is invalid. Maybe we should log only from 1 place (e.g. from controllers only)?